### PR TITLE
Create wrappers over the DiscordRequest classes

### PR DESCRIPTION
**Pull request by surajkumar**
Created a bunch of wrappers over the DiscordRequest classes. 

TODOs:

- Finish the Channel requests
- Add the Guild requests

### CODE REVIEW

File: models/src/main/java/com/javadiscord/jdi/core/models/message/StickerPack.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.util.List;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+public record StickerPack(
+        long id,
+        List<Sticker> stickers,
+        String name,
+        long sku_id,
```
Comment (nateweisz) [inline models/src/main/java/com/javadiscord/jdi/core/models/message/StickerPack.java:13]: Need's to use @JsonProperty and rename to skuId

File: api/src/main/java/com/javadiscord/jdi/core/api/builders/ModifyStageBuilder.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
+package com.javadiscord.jdi.core.api.builders;
+
+import com.javadiscord.jdi.internal.api.stage.ModifyStageRequest;
+
+import java.util.Optional;
+
+public class ModifyStageBuilder {
+    private final String channelId;
```
Comment (Taz03) [inline api/src/main/java/com/javadiscord/jdi/core/api/builders/ModifyStageBuilder.java:8]: channelId string?

File: api/src/main/java/com/javadiscord/jdi/core/api/utils/DiscordImageUtil.java [hunk 3]
```diff
@@ -8,8 +8,7 @@
 import java.util.List;
 
 public class DiscordImageUtil {
-    private static final List<String> IMAGE_EXTENSIONS =
-            Arrays.asList("jpg", "jpeg", "png", "gif", "bmp");
+    private static final List<String> IMAGE_EXTENSIONS = Arrays.asList("jpg", "jpeg", "png", "gif");
```
Comment (Taz03) [inline api/src/main/java/com/javadiscord/jdi/core/api/utils/DiscordImageUtil.java:11]: um, this is out of scope but, List.of ?

File: models/src/main/java/com/javadiscord/jdi/core/models/guild/Onboarding.java [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+import java.util.List;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+public record Onboarding(
+        @JsonProperty("guild_id") long guildId,
+        List<OnboardingPrompt> prompts,
+        @JsonProperty("default_channel_ids") List<Long> defaultChannelIds,
+        boolean enabled,
+        OnboardingMode mode) {}
```
Comment (Taz03) [inline models/src/main/java/com/javadiscord/jdi/core/models/guild/Onboarding.java:14]: @JsonProperty

File: models/src/main/java/com/javadiscord/jdi/core/models/guild/PruneCount.java [hunk 5]
```diff
@@ -0,0 +1,6 @@
+package com.javadiscord.jdi.core.models.guild;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+public record PruneCount(int pruned) {}
```
Comment (Taz03) [inline models/src/main/java/com/javadiscord/jdi/core/models/guild/PruneCount.java:6]: @JsonProperty

File: models/src/main/java/com/javadiscord/jdi/core/models/voice/VoiceRegion.java [hunk 6]
```diff
@@ -0,0 +1,7 @@
+package com.javadiscord.jdi.core.models.voice;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+public record VoiceRegion(
+        String id, String name, boolean optimal, boolean deprecated, boolean custom) {}
```
Comment (Taz03) [inline models/src/main/java/com/javadiscord/jdi/core/models/voice/VoiceRegion.java:7]: @JsonProperty

File: api/src/main/java/com/javadiscord/jdi/core/api/builders/CreateMessageBuilder.java [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+    private final long channelId;
+    private Optional<String> content;
+    private Optional<Integer> nonce;
+    private Optional<Boolean> tts;
+    private Optional<List<Embed>> embeds;
+    private Optional<List<ChannelMention>> mentions;
+    private Optional<List<MessageReference>> messageReferences;
+    private Optional<List<Object>> components; // TODO: Create Components object
+    private Optional<List<Long>> stickerIds;
+    private Optional<Object> files; // TODO: Create File object
```
Comment (Taz03) [inline api/src/main/java/com/javadiscord/jdi/core/api/builders/CreateMessageBuilder.java:23]: Make models before merging

File: api/src/main/java/com/javadiscord/jdi/core/api/builders/StartThreadInForumOrMediaChannelBuilder.java [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+import java.util.Optional;
+
+public class StartThreadInForumOrMediaChannelBuilder {
+    private final long channelId;
+    private final String name;
+    private Optional<Integer> autoArchiveDuration;
+    private Optional<Integer> rateLimitPerUser;
+    private final Object message; // TODO: Create Forum and Media Thread Message Params Object
+    private Optional<List<Long>> appliedTags;
+    private Optional<Object> files; // TODO: Create Files object
```
Comment (Taz03) [inline api/src/main/java/com/javadiscord/jdi/core/api/builders/StartThreadInForumOrMediaChannelBuilder.java:15]: create models before merging

**Review by Taz03 — CHANGES_REQUESTED** [review]

File: api/src/main/java/com/javadiscord/jdi/core/api/builders/ExecuteGithubCompatibleWebhookRequestBuilder.java [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+        this.threadId = Optional.empty();
+        this.waits = Optional.empty();
+    }
+
+    public ExecuteGithubCompatibleWebhookRequestBuilder threadId(Optional<Long> threadId) {
+        this.threadId = threadId;
+        return this;
+    }
+
+    public ExecuteGithubCompatibleWebhookRequestBuilder waits(Optional<Boolean> waits) {
```
Comment (Taz03) [inline api/src/main/java/com/javadiscord/jdi/core/api/builders/ExecuteGithubCompatibleWebhookRequestBuilder.java:25]: fix

File: api/src/main/java/com/javadiscord/jdi/internal/api/webhook/EditWebhookMessageRequest.java [hunk 10]
```diff
@@ -16,9 +16,9 @@ public record EditWebhookMessageRequest(
         Optional<Long> threadId,
         Optional<String> content,
         Optional<List<Embed>> embeds,
-        Optional<Object> allowedMentions,
+        Optional<Object> allowedMentions, // TODO: Create Model
         Optional<List<Object>> components,
-        Optional<Object> files,
+        Optional<Object> files, // TODO: Create Model
```
Comment (Taz03) [inline api/src/main/java/com/javadiscord/jdi/internal/api/webhook/EditWebhookMessageRequest.java:21]: make models before merging

File: api/src/main/java/com/javadiscord/jdi/internal/api/webhook/ExecuteWebhookRequest.java [hunk 11]
```diff
@@ -21,10 +22,10 @@ public record ExecuteWebhookRequest(
...
         Optional<String> avatarUrl,
         Optional<Boolean> tts,
-        Optional<List<Object>> embeds,
-        Optional<Object> allowedMentions,
-        Optional<List<Object>> components,
-        Optional<Object> files,
+        Optional<List<Embed>> embeds,
+        Optional<Object> allowedMentions, // TODO: Create model
+        Optional<List<Object>> components, // TODO: Create model
+        Optional<Object> files, // TODO: Create model
```
Comment (Taz03) [inline api/src/main/java/com/javadiscord/jdi/internal/api/webhook/ExecuteWebhookRequest.java:28]: make models before merging

**Review by Taz03 — CHANGES_REQUESTED** [review]

**Pull request merged**